server/src/weaverbird/pipeline/pipeline.py

```diff
@@ -256,6 +256,74 @@ def remove_void_conditions_from_filter_steps(
     return final_steps
 
 
+def _remove_empty_elements(
+    data: dict[str, Any] | list[Any] | None
+) -> dict[str, Any] | list[Any] | None:
```

Comment:
I believe this signature is wrong: You're calling the function on the values of a `dict[str, Any]`. Therefore, it should accept `Any`

```suggestion
def _remove_empty_elements(data: Any) -> Any:
```